File: students/k3343/Otroschenko_Valeria/Lr1/task-3/HttpServer.cs
using System.Net;
using System.Net.Sockets;
using System.Text;

namespace ConsoleApp1.task_3;

class HttpServer
{
    private const string IpAddress = "127.0.0.1";
    private const int Port = 8080;
    private const int BufferSize = 1024;
    private const int Offset = 0;
    private const string HtmlPath = "/Users/valeriaotrosenko/RiderProjects/ConsoleApp1/ConsoleApp1/task-3/index.html";
    private const string HttpResponseHeader = "HTTP/1.1 200 OK\r\n" +
                                              "Content-Type: text/html; charset=UTF-8\r\n";

    public static void Main()
    {
        IPAddress ip = IPAddress.Parse(IpAddress);
        TcpListener listener = new TcpListener(ip, Port);
        listener.Start();
        Console.WriteLine($"Сервер запущен на {ip}:{Port}. Ожидание подключений...");

        while (true)
        {
            TcpClient client = listener.AcceptTcpClient();
            Console.WriteLine("Клиент подключен.");

            NetworkStream stream = client.GetStream();

            byte[] buffer = new byte[BufferSize];
            int bytesRead = stream.Read(buffer, Offset, buffer.Length);
            string request = Encoding.UTF8.GetString(buffer, Offset, bytesRead);
            Console.WriteLine("Запрос от клиента:\n" + request);

            string htmlContent = File.ReadAllText(HtmlPath);

            string httpResponse = HttpResponseHeader +
                                  $"Content-Length: {htmlContent.Length}\r\n" +
                                  "\r\n" +
                                  htmlContent;

            byte[] responseBytes = Encoding.UTF8.GetBytes(httpResponse);
            stream.Write(responseBytes, Offset, responseBytes.Length);

            stream.Close();
            client.Close();
        }
    }
}

File: students/k3341/laboratory_works/Цветкова_Татьяна/laboratory_work_1/socket_clean/http_server.py
import socket

HOST = "127.0.0.1"
PORT = 8080

server = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
server.bind((HOST, PORT))
server.listen(5)

print(f"HTTP сервер на http://{HOST}:{PORT}")

while True:
    conn, _ = server.accept()
    conn.recv(1024)

    with open("index.html", "r", encoding="utf-8") as f:
        html = f.read()

    response = (
        "HTTP/1.1 200 OK\r\n"
        "Content-Type: text/html; charset=utf-8\r\n\r\n"
        + html
    )

    conn.send(response.encode("utf-8"))
    conn.close()